### PR TITLE
add hardwaretier validation; adjust run logging

### LINKED ISSUE #67: add hardware tier validation check to domino operator
Now that hardware tiers can be enumerated per project via the API it would be nice if the `DominoOperator` validated that the tier was valid before starting a run and threw a proper exception if invalid (instead of the typical `JSONDecodeError`).

### PR DESCRIPTION
**Pull request by wseaton**
fixes #67, updates logging in accordance with changes from #69.

The way that the `runs_start_blocking` method logs through the exception msg made it hard to catch and handle cleanly, instead I moved around the log processor from the Domino operator and re-factored it to use the new `.log` property which plays nice with airflow. 

Open to suggestions if you guys don't like this behavior, the cost of not fixing it is ugly formatting for logs on airflow job failure, since a pretty printed string is what is passed through the exception (which I then have to de-pretty print), I think ideally we'd get a raw log.

### CODE REVIEW

File: domino/airflow/_operator.py [hunk 1]
```diff
@@ -70,14 +69,25 @@ def __init__(
         self.startup_delay = startup_delay
         self.include_setup_log = include_setup_log
 
-        self.client = None
-        self.run_id = None
+        self.client: Optional[Domino] = None
```
Comment (abhijeet2096) [inline domino/airflow/_operator.py:72]: 👍

File: domino/airflow/_operator.py [hunk 2]
```diff
@@ -2,11 +2,10 @@
 import logging
```
Comment (abhijeet2096) [inline domino/airflow/_operator.py:2]: No longer needed.

File: domino/airflow/_operator.py [hunk 3]
```diff
@@ -2,11 +2,10 @@
...
 from typing import List, Optional, Any
 
-from bs4 import BeautifulSoup
-
 from airflow.models import BaseOperator
 from airflow.utils.decorators import apply_defaults
 from domino import Domino
+from domino.exceptions import RunFailedException
```
Comment (abhijeet2096) [inline domino/airflow/_operator.py:8]: unused import

File: tests/test_operator.py [hunk 4]
```diff
@@ -30,3 +30,42 @@ def test_my_operator():
     )
     ti = TaskInstance(task=task, execution_date=datetime.now())
     task.execute(ti.get_template_context())
+
```
Comment (abhijeet2096) [inline tests/test_operator.py:33]: Add blank line before `test_operator_fail` and `test_operator_fail_invalid_tier` and at end of file

File: tests/test_operator.py [hunk 5]
```diff
@@ -1,20 +1,20 @@
...
+local mode and run:
+
+`airflow initdb`
+"""
 import os
 from datetime import datetime
 
 import pytest
-
+import logging
```
Comment (abhijeet2096) [inline tests/test_operator.py:11]: unused import

**Review by abhijeet2096 — APPROVED** [review]
Looks Good, Please fix minor comments

**Comment by wseaton** [comment]
@abhijeet2096 Style changes have been made

**Pull request merged**